crates/bevy_asset/src/path.rs

```diff
@@ -63,6 +63,51 @@ impl<'a> AssetPath<'a> {
                 .map(|value| Cow::Owned(value.to_string())),
         }
     }
+
+    /// Resolves a relative asset path. The result will be an `AssetPath` which is resolved relative
+    /// to this path. There are three cases:
+    ///
+    /// If the relative path begins with `#`, then it is considered an asset label, in which case
+    /// the result is the base path with the label portion replaced.
+    ///
+    /// If the relative path starts with "./" or "../", then the result is combined with the base
```

Comment:
Another nice feature would be to resolve paths relative to the the same directory of the current `AssetPath`. So rather than specifying `./foo.png` or `./foo/bar.txt`, users only need to put `foo.png` or `foo/bar.txt` (where the `./` is implied).

This would mean that the given `relative_path` would pretty much always need to be considered relative. In [`bevy_proto`](https://github.com/MrGVSV/bevy_proto/blob/37a72d719309f97e7b50a14fb4997d3693e874cf/bevy_proto_backend/src/path/proto_path.rs#L17-L26), I made it so that a leading `/` denoted a path relative to the asset root, but I don't think that has to be the case here since this method is meant for relative paths.

Reply:
BTW, if I were going to do as you suggest, I wouldn't call this method `resolve`, I would call it `join` or `concat` or something along those lines.

The design is inspired by the node.js function [path.resolve()](https://nodejs.org/api/path.html#pathresolvepaths), although that function is more complicated and is much more of a swiss army knife than this one is (for one thing, it takes an arbitrary number of arguments).

Reply:
I see. In that case, maybe we can split the join functionality you already have here into a dedicated `AssetPath::join` method? Or maybe that can happen as a followup PR?

I would have suggested the `/` trick that I mentioned earlier, but I don't know how well that fits in the context of Bevy's `AssetPath` proper.

---

Also, I would suggest renaming `relative_path` to `path` or something similar. The `relative` might throw off users who think that the given path will always be relative to the base path, even though that's not the case.

Reply:
First let me say - you're not wrong.

Relative URIs show up in a lot of places: clicking on hyperlinks, XML XPointers, JSON-Schema files, and others I could name. (Thus the word "Universal" in the name). There isn't a consistent treatment as to how these are resolved, however it generally is one of two choices, which are the two choices under discussion here. For example, the behavior of clicking on a relative link is different than the behavior of `new URL(relative_path, base_path)` in JavaScript. For a link, a relative path that starts with a name, and not a special character, is resolved relative to the document root. For `new URL()`, it's resolved relative to the base path.

For Bevy, I think the decision should be whatever is going to be the most useful for the common use cases we anticipate. And also to make sure that we document the behavior clearly.

I'm open to adding a `join` method in a subsequent PR.

Reply:
I have renamed the argument as you have requested. I have also re-written the doc comment a bit more to hopefully clarify some potential ambiguities.